### PR TITLE
Add OrderedMessage and OrderedEvent to traces & txn state diffs

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

**Comment by omerfirmak** [comment]
Closes #1228

File: vm/rust/src/lib.rs [hunk 1]
```diff
@@ -223,10 +221,13 @@ pub extern "C" fn cairoVMExecute(
             return;
         }
 
+        let mut txn_state = CachedState::create_transactional(&mut state);
```
Comment (kirugan) [inline vm/rust/src/lib.rs:224]: Could you explain why do we need this?

File: vm/rust/src/jsonrpc.rs [hunk 2]
```diff
@@ -32,7 +39,35 @@ pub struct TransactionTrace {
...
+                _ => {}
+            }
+        }
+        if self.fee_transfer_invocation.is_some() {
+            self.fee_transfer_invocation.as_mut().unwrap().make_legacy()
+        }
+        if self.constructor_invocation.is_some() {
+            self.constructor_invocation.as_mut().unwrap().make_legacy()
+        }
+        if self.function_invocation.is_some() {
```
Comment (kirugan) [inline vm/rust/src/jsonrpc.rs:67]: common rust idiom is to use if let in such cases i.e. 
```
if let Some(actualValue) = self.function_invocation {
   // do something with
}
```
but it's up to you

Comment (omerfirmak) [reply vm/rust/src/jsonrpc.rs]: I knew there was a better way of doing this:)

File: vm/rust/src/lib.rs (hunk 1 above)
Comment (omerfirmak) [reply vm/rust/src/lib.rs:224]: to be able to generate per-txn state diffs. 

txn_state buffers the changes internally, until `commit()` is called. So we gather all the changes for a single txn, save them separately and then commit them to `state`.

**Review by kirugan — APPROVED** [review]

**Pull request merged**